File: Android/ui/src/main/java/com/felipearpa/tyche/ui/FlagImage.kt
package com.felipearpa.tyche.ui

import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import coil3.compose.AsyncImage

@Composable
fun FlagImage(
    teamId: String,
    modifier: Modifier = Modifier,
    contentDescription: String? = null,
) {
    flagResource(teamId)?.let { resourceId ->
        AsyncImage(
            model = resourceId,
            contentDescription = contentDescription,
            modifier = modifier,
        )
    }
}

private fun flagResource(teamId: String): Int? = when (teamId.lowercase()) {
    "ar" -> R.drawable.flag_ar
    "at" -> R.drawable.flag_at
    "au" -> R.drawable.flag_au
    "ba" -> R.drawable.flag_ba
    "be" -> R.drawable.flag_be
    "br" -> R.drawable.flag_br
    "ca" -> R.drawable.flag_ca
    "cd" -> R.drawable.flag_cd
    "ch" -> R.drawable.flag_ch
    "ci" -> R.drawable.flag_ci
    "co" -> R.drawable.flag_co
    "cv" -> R.drawable.flag_cv
    "cw" -> R.drawable.flag_cw
    "cz" -> R.drawable.flag_cz
    "de" -> R.drawable.flag_de
    "dz" -> R.drawable.flag_dz
    "ec" -> R.drawable.flag_ec
    "eg" -> R.drawable.flag_eg
    "es" -> R.drawable.flag_es
    "fr" -> R.drawable.flag_fr
    "gb_eng" -> R.drawable.flag_gb_eng
    "gb_sct" -> R.drawable.flag_gb_sct
    "gh" -> R.drawable.flag_gh
    "hr" -> R.drawable.flag_hr
    "ht" -> R.drawable.flag_ht
    "iq" -> R.drawable.flag_iq
    "ir" -> R.drawable.flag_ir
    "jo" -> R.drawable.flag_jo
    "jp" -> R.drawable.flag_jp
    "kr" -> R.drawable.flag_kr
    "ma" -> R.drawable.flag_ma
    "mx" -> R.drawable.flag_mx
    "nl" -> R.drawable.flag_nl
    "no" -> R.drawable.flag_no
    "nz" -> R.drawable.flag_nz
    "pa" -> R.drawable.flag_pa
    "pt" -> R.drawable.flag_pt
    "py" -> R.drawable.flag_py
    "qa" -> R.drawable.flag_qa
    "sa" -> R.drawable.flag_sa
    "se" -> R.drawable.flag_se
    "sn" -> R.drawable.flag_sn
    "tn" -> R.drawable.flag_tn
    "tr" -> R.drawable.flag_tr
    "us" -> R.drawable.flag_us
    "uy" -> R.drawable.flag_uy
    "uz" -> R.drawable.flag_uz
    "za" -> R.drawable.flag_za
    else -> null
}

File: iOS/UI/Sources/UI/Assets/FlagImage.swift
import SwiftUI

public struct FlagImage: View {
    private let teamCode: String

    public init(teamCode: String) {
        self.teamCode = teamCode
    }

    public var body: some View {
        if let resource = flagResource(for: teamCode) {
            Image(sharedResource: resource)
                .resizable()
                .scaledToFit()
        }
    }
}

private func flagResource(for teamCode: String) -> SharedImageResource? {
    switch teamCode.lowercased() {
    case "ar": return SharedImageResource(.flagAr)
    case "at": return SharedImageResource(.flagAt)
    case "au": return SharedImageResource(.flagAu)
    case "ba": return SharedImageResource(.flagBa)
    case "be": return SharedImageResource(.flagBe)
    case "br": return SharedImageResource(.flagBr)
    case "ca": return SharedImageResource(.flagCa)
    case "cd": return SharedImageResource(.flagCd)
    case "ch": return SharedImageResource(.flagCh)
    case "ci": return SharedImageResource(.flagCi)
    case "co": return SharedImageResource(.flagCo)
    case "cv": return SharedImageResource(.flagCv)
    case "cw": return SharedImageResource(.flagCw)
    case "cz": return SharedImageResource(.flagCz)
    case "de": return SharedImageResource(.flagDe)
    case "dz": return SharedImageResource(.flagDz)
    case "ec": return SharedImageResource(.flagEc)
    case "eg": return SharedImageResource(.flagEg)
    case "es": return SharedImageResource(.flagEs)
    case "fr": return SharedImageResource(.flagFr)
    case "gb_eng": return SharedImageResource(.flagGbEng)
    case "gb_sct": return SharedImageResource(.flagGbSct)
    case "gh": return SharedImageResource(.flagGh)
    case "hr": return SharedImageResource(.flagHr)
    case "ht": return SharedImageResource(.flagHt)
    case "iq": return SharedImageResource(.flagIq)
    case "ir": return SharedImageResource(.flagIr)
    case "jo": return SharedImageResource(.flagJo)
    case "jp": return SharedImageResource(.flagJp)
    case "kr": return SharedImageResource(.flagKr)
    case "ma": return SharedImageResource(.flagMa)
    case "mx": return SharedImageResource(.flagMx)
    case "nl": return SharedImageResource(.flagNl)
    case "no": return SharedImageResource(.flagNo)
    case "nz": return SharedImageResource(.flagNz)
    case "pa": return SharedImageResource(.flagPa)
    case "pt": return SharedImageResource(.flagPt)
    case "py": return SharedImageResource(.flagPy)
    case "qa": return SharedImageResource(.flagQa)
    case "sa": return SharedImageResource(.flagSa)
    case "se": return SharedImageResource(.flagSe)
    case "sn": return SharedImageResource(.flagSn)
    case "tn": return SharedImageResource(.flagTn)
    case "tr": return SharedImageResource(.flagTr)
    case "us": return SharedImageResource(.flagUs)
    case "uy": return SharedImageResource(.flagUy)
    case "uz": return SharedImageResource(.flagUz)
    case "za": return SharedImageResource(.flagZa)
    default: return nil
    }
}
